benchexec/tools/jcwit.py

```diff
@@ -0,0 +1,69 @@
+# This file is part of BenchExec, a framework for reliable benchmarking:
+# https://github.com/sosy-lab/benchexec
+#
+# SPDX-FileCopyrightText: 2007-2020 Dirk Beyer <https://www.sosy-lab.org>
+#
+# SPDX-License-Identifier: Apache-2.0
+
+import benchexec.tools.template
+import benchexec.result as result
+from benchexec.tools.template import ToolNotFoundException
+
+
+class Tool(benchexec.tools.template.BaseTool2):
+
+    """
+    Tool info for jcwit
+    (https://github.com/Chriszai/JCWIT).
+    """
+
+    def executable(self, tool_locator):
+        return tool_locator.find_executable("jcwit.py")
+
+    def version(self, executable):
+        return self._version_from_tool(executable)
+
+    def name(self):
+        return "jcwit"
+
+    def cmdline(self, executable, options, task, rlimits):
+        return [executable] + options + list(task.input_files)
+
+    def determine_result(self, run):
+        output = run.output
+        validation = "unknown"
+        for line in output:
+            if "Exception" in line:
+                if "AssertionError" in line:
+                    validation = "false"
+                else:
+                    validation = "unknown"
+                    
+            if "Witness result: Unknown" in line:
+                validation = "unknown"
+                break

```

Comment:
Setting a variable to a string here, then `break` and then compare that variable to the string again in order to set another variable to another value just in order to return that last value is overly complex and makes the control flow here difficult to understand. Why not just do `return result.RESULT_FALSE_PROP`? (Same for the similar cases below.)

If you want to make sure that having `Exception` in the output returns in an error result (which is understandable), then the current logic also does not fulfill this and needs to be changed (no `break`s for example).